airflow/www/views.py

```diff
@@ -3302,7 +3302,7 @@ class DagRunModelView(AirflowModelView):
         'external_trigger',
         'conf',
     ]
-    edit_columns = ['state', 'dag_id', 'execution_date', 'run_id', 'conf']
+    edit_columns = ['state', 'dag_id', 'execution_date', 'start_date', 'end_date', 'run_id', 'conf']
```

Comment:
May you clarify a bit more about this change? Thanks.

Reply:
I'm not sure why we need this change to address the issue.

Reply:
I see this in the PR description but not sure if I understand it -- when you mark a DagRun to state the Scheduler should set the appropriate date instead of the user

>An issue with the DagRunEditForm, which would clear the start_date and end_date for a DagRun, if the form was used to transition a DagRun from a failed state back into a running state (or any other state). In the event where the scheduler would determine the DagRun should've been in the failed or success state (e.g. because the task instances weren't cleared), then this would lead to a scheduler crash.

Reply:
Sure! Happy to clarify why I choose to venture down this route.

The [`DagRunEditForm`](https://github.com/apache/airflow/blob/master/airflow/www/forms.py#L172-L173) contains the `start_date` and `end_date` fields, and these are supposed to be "read-only" as they use [`AirflowDateTimePickerROWidget`](https://github.com/apache/airflow/blob/45e72ca83049a7db526b1f0fbd94c75f5f92cc75/airflow/www/widgets.py#L51).

However, what would happen when I would use the `DagRunEditForm` through the Web UI is that the `start_date` and `end_date` would be set to `NULL` when submitting the form (even if they were set previously). This happened because `start_date` and `end_date` are not contained in the `edit_columns` list, and thus the form would not contain any data for them and overwrite old values.

By adding `start_date` and `end_date` as entries to `edit_columns` columns, the widget displays them as "read-only" in the form, and no longer nulls them out when the form is submitted.

Note that any DagRun that is in the `running` state, but does not have a `start_date` set may cause a scheduler crash. This crash is avoided by https://github.com/apache/airflow/pull/14452/commits/a780e8a029c3801cd60da784fcf0b7af2c58bf13.

Reply:
Thanks @RikHeijdens for the details.

Makes sense to me, and also looks good to me after I tested in a local minimized setup.

I will approve. Meanwhile @kaxil may you also share your inputs or any question to further clarify?

Reply:
Oh yes I forgot -- I had added the Read-only Widgets for TI and DagRuns in https://github.com/apache/airflow/pull/12770 to fix another bug

Reply:
![image](https://user-images.githubusercontent.com/8811558/100955885-4991b700-350f-11eb-9a32-ce199f530d6e.gif)